### ISSUE TITLE
Imitation learning crashes if state size is zero

### ISSUE THREAD

**Issue by Lakrix** [Unity-Technologies/ml-agents#540]
trainer.py crashes if only camera is used as observation and the number of state is defined as zero. Need to do a check if bc/trainer.py if self.use_states:

**Comment by akshitac8** [comment]
./

**Comment by Lakrix** [comment]
I missed that there is possibility to have camera observation during imitation learning. Will this be added in future releases?

**Comment by RossMelbourne** [comment]
I was able to make 'visual observations only' work but did have to pass some dummy state data because I received the same crash.  In addition to dummy state data, I was able to get imitation learning to work by using a very small visual observation width and height (120 x 160) and also skipping 10 frames before doing any learning (Decision Frequency on Agent set to 10). I set Quality Level to 1.  Also, I ignored the message 'Cannot use observations with imitation learning' after examining the python code and suspecting it would work.  I am assuming it is not supported because sending visual observations to python is cpu intensive but I am not sure if that is where the bottle neck might be.
